src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -509,6 +516,19 @@ private void ParseHelper(string[] args)
                     switchKey = switchKey.Substring(1);
                 }
 
+                // If version is in the commandline, don't continue to look at any other parameters
+                if (MatchSwitch(switchKey, "version", "v"))
+                {
+                    _showVersion = true;
+                    _showBanner = false;
+                    _noInteractive = true;
+                    _skipUserInit = true;
+                    _noExit = false;
+                    _commandLineCommand = "'powershell ' + $psversiontable.gitcommitid";
```

Comment:
vestige of earlier implementation - removed

---
In reply to: [100401276](https://github.com/PowerShell/PowerShell/pull/3115#discussion_r100401276) [](ancestors = 100401276)